### PR TITLE
feat: introduce function to generate url with optional params

### PR DESCRIPTION
**Pull request by ikondrat**
## What has changed and why?

We need to use more query parameters for urls for routes and add it without proper handling function would lead to the mess in the code. This PR introduces function allowing to generate url with optional query parameters in aligned way.

It will boilerplate to worj with query parameters

## How has it been tested?

By unit tests

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

## Summary

Introduces a new internal utility getURL to centralize and standardize URL generation with optional query parameters, reducing boilerplate and inconsistent query handling across route helpers. The implementation omits undefined, null, and empty-string values when building query strings and returns the base path unchanged when there are no query parameters.

## Changes

- New utility: lightly_studio_view/src/lib/utils/getUrl/getUrl.ts
  - Exported function: getURL(basePath: string, params?: Record<string, string | number | boolean | undefined>): string
  - Builds a query string via URLSearchParams, omits values that are undefined, null, or empty strings, converts values to strings, and returns basePath with ?query if non-empty.
  - Query values are URL-encoded.

- Tests: lightly_studio_view/src/lib/utils/getUrl/getUrl.test.ts
  - Vitest suite covering: returning basePath unchanged when no params, appending query params, omitting undefined/null/empty-string params, preserving numeric zero, and correct URL encoding (including spaces and reserved characters).

- Re-exports: lightly_studio_view/src/lib/utils/index.ts
  - Added named export getURL.

- Refactors: lightly_studio_view/src/lib/routes.ts
  - routes.collection.videosDetails(...) and routes.collection.framesDetails(...) now delegate URL/query construction to getURL instead of manually assembling query strings; existing conditional logic for including frame_number and from_video is preserved.

## Notes

- No change to exported signatures of existing route helpers.
- Unit tests added to cover getURL behavior.
- CHANGELOG.md not updated (internal change).
- Code-style guidelines in AGENTS.md were reviewed; no style-related changes required by this PR.

### CODE REVIEW

File: lightly_studio_view/src/lib/utils/getURL/getURL.test.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { describe, it, expect } from 'vitest';
+import { getURL } from '$lib/utils';
+
+describe('getURL', () => {
```
Comment (LeonardoRosaa) [inline lightly_studio_view/src/lib/utils/getURL/getURL.test.ts:4]: I'd like to see a test with an empty field. For example:

```
const result = getURL('/api/data', { id: '', page: 1 })
```

Comment (ikondrat) [reply lightly_studio_view/src/lib/utils/getURL/getURL.test.ts:4]: updated

Comment (LeonardoRosaa) [reply lightly_studio_view/src/lib/utils/getURL/getURL.test.ts:4]: I'd also test a zero field. For example:

const result = getURL('/api/data', { id: '', page: 0 })

Comment (ikondrat) [reply lightly_studio_view/src/lib/utils/getURL/getURL.test.ts:4]: added test for 0 values as well
